### PR TITLE
fix: do not forget leading zeroes when decoding

### PR DESCRIPTION
**Pull request by geonnave**
Solution to #27. Included a few tests and it worked. Inspired by [this](https://github.com/keis/base58/blob/master/base58/__init__.py#L131).

### CODE REVIEW

File: lib/base58.ex [hunk 1]
```diff
@@ -36,11 +36,23 @@ defmodule Base58 do
...
+  def decode(binary) do
+    {zeroes, binary} = handle_leading_zeroes(binary)
+    zeroes <> decode(binary, 0)
+  end
+  def decode("", 0), do: ""
   def decode("", acc), do: :binary.encode_unsigned(acc)
   def decode(<<head, tail::binary>>, acc),
     do: decode(tail, acc * 58 + Enum.find_index(@alnum, &(&1 == head)))
 
+  def handle_leading_zeroes(binary) do
```
Comment (nelsonic) [inline lib/base58.ex:48]: This function probably doesn't need to be exposed externally. Should we make it a `defp`? 💭 
```suggestion
  defp handle_leading_zeroes(binary) do
```

Comment (geonnave) [reply lib/base58.ex]: Good catch, done.

**Review by nelsonic — APPROVED** [review]
@geonnave thanks for reporting this issue and proactively addressing it. 🙌

**Comment by nelsonic** [comment]
Package published to https://hex.pm/packages/b58/1.0.3 contains this update. 📦 
Thanks again! 🥇

**Comment by geonnave** [comment]
Happy to help! 🙌

**Comment by nelsonic** [comment]
Invited you to be a maintainer: https://github.com/dwyl/base58/invitations + https://github.com/dwyl as you clearly have the skills & attention to detail. 100% optional. Know you're busy with your PhD research. 👍 
Hope you have a great Friday & Weekend. ☀️

**Comment by geonnave** [comment]
Thanks @nelsonic, I've accepted the invitation for `base58`. Let me learn a bit about `dwyl` :)

**Pull request merged**